ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryIdConverterFactory.java

```diff
@@ -208,6 +209,23 @@ private CompletionStage<String> convertId(String input, RestRequest restRequest,
       return conversionFuture;
     }
 
+    /**
+     * @param restRequest incoming {@link RestRequest}
+     * @return {@code true} if this request needs to be inserted into named blob DB
+     */
+    private boolean isNamedBlobDBPutRequest(RestRequest restRequest) {
```

Comment:
Feels like we need a better name to cover both cases.

Reply:
Or we just separate them to two conditions, isNamedBlobPutRequest() || isS3MultipartCompleteRequest()

Reply:
Sure. I split the naming to `isNamedBlobPutRequest() || isS3MultipartCompleteRequest()`